### PR TITLE
Added a getName() function for Trees, added a little more documentation.

### PR DESCRIPTION
**Pull request by Sandertv**
This pull request is nothing special. It makes getting the name of Tree objects a bit easier, which is useful for plugins at times. I also added a bit more documentation. It seemed a bit unclear to me that growing trees didn't work if you grew them from a dirt block, but it should be the block above it.

### CODE REVIEW

**Comment by Sandertv** [comment]
I'm trying to fix the spaces that should be before the : , but git won't let me push those...

**Comment by dktapps** [comment]
May I ask what editor you are using?

**Comment by Sandertv** [comment]
PHPStorm. Just found out that it had the auto reformat code option on when making a commit. Sorry for that.

**Comment by dktapps** [comment]
There's still trailing whitespace everywhere, would you mind fixing it please?

**Comment by dktapps** [comment]
Also, you could use [this](https://github.com/pmmp/PocketMine-MP/pull/485/files#diff-d35e4fcc44efbe13605fefcbdf2a5373R40) and check it against wood block names instead of hardcoding it. I don't like hardcoding stuff.

**Comment by Sandertv** [comment]
Fixing trailing whitespace right now... Problem for the wood checking though, is that it will get messy when large trees get implemented. Any ideas on that?

**Comment by Sandertv** [comment]
Actually, I just thought of something that should do the job.

File: src/pocketmine/level/generator/object/BirchTree.php [hunk 1]
```diff
@@ -35,6 +35,7 @@ public function __construct($superBirch = false){
 		$this->leafBlock = Block::LEAVES;
 		$this->type = Wood::BIRCH;
 		$this->superBirch = (bool) $superBirch;
+		$this->isBig = (bool)$superBirch;
```
Comment (Sandertv) [inline src/pocketmine/level/generator/object/BirchTree.php]: *facepalm* 😫

**Comment by Sandertv** [comment]
Okay this can be merged now, if there's no more things that have to be changed.

**Comment by Sandertv** [comment]
I've just accidentally pushed something that should be put in a separated pull request to this one. It's not an actual change, but I'll find a way to fix it if you don't want to merge it like this.

**Comment by dktapps** [comment]
Easily fixed. Just cherry-pick the commit onto a different branch, and then:

```
git reset HEAD~1 --hard
git push -f
```

That will reset the branch back by 1 commit and force-update your fork. Make sure you cherry-pick the offending commit somewhere else or you will lose it.

**Comment by Sandertv** [comment]
Obsoleted by https://github.com/pmmp/PocketMine-MP/pull/815

**Pull request closed without merging**